src/tunnel.cc

```diff
@@ -224,6 +223,9 @@ class TunnelStateData: public PeerSelectionInitiator
     void saveError(ErrorState *finalError);
     void sendError(ErrorState *finalError, const char *reason);
 
+    /// the processing tags associated with this request transaction.
+    LogTags &logType() { return al->cache.code; }
```

Comment:
If `al` is indeed always available, then I think this method should be removed (i.e. not added) in favor of direct access to ALE.

Please note that there is at least one `if (al)` in the official code. If this PR claims that ALE is always there, then we should remove that code unconditional _and_ add an explicit "We have verified that TunnelStateData::al is always available." claim to the PR description.